src/couch/src/couch_lists.erl

```diff
@@ -0,0 +1,46 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_lists).
+
+-export([uniq/1]).
+
+%% uniq/1: return a new list with the unique elements of the given list
+
+-spec uniq(List1) -> List2 when
+    List1 :: [T],
+    List2 :: [T],
+    T :: term().
+
+-ifdef(OTP_RELEASE).
+-if((?OTP_RELEASE) >= 25).
+-define(HAS_LISTS_UNIQ, true).
```

Comment:
why not put the code here (and the impl in an else)? the extra set of pragmas don't add anything afaict.

Reply:
These are two nested `if`s where each could have its own `else` branches.  This creates 3 possible branches: 1 for using `lists:uniq/1` directly and 2 for using its imported definition (no `OTP_RELEASE` defined or it is defined but its value is lesser than `25`).  I wanted to avoid the code duplication for these latter ones.

I know `HAS_LISTS_UNIQ` may look `autoconf`ish but it works nicely.  For what it is worth, I was inspired by [`mochiweb`](https://github.com/mochi/mochiweb/blob/073349423408e4bae08c90f8d16fee5317e45263/examples/example_project/src/example_project_web.erl#L22-L32).

Reply:
`?OTP_RELEASE` was introduced in Erlang/OTP 21 and our minimum version is now 23, so we don't need to test for it. I see recent commits where we've removed the extra `-ifdef(OTP_RELEASE)` check. I think remove that and seeing if we can simplify?

Reply:
Sure, that is also an option for me.